test/zdtm.py

```diff
@@ -513,8 +513,14 @@ def kill(self, sig=signal.SIGKILL):
         self.__freezer.thaw()
         if self.__pid:
             print("Send the %d signal to  %s" % (sig, self.__pid))
-            os.kill(int(self.__pid), sig)
-            self.gone(sig == signal.SIGKILL)
+            try:
+                os.kill(int(self.__pid), sig)
+            except ProcessLookupError:
+                # Prevent gone() calling waitpid().
+                self.auto_reap = True
```

Comment:
if auto_reap is false, we need to raise the exception, because this pid can't disappear before we call waitpid.